src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py

```diff
@@ -223,8 +223,9 @@ def _construct_obj_for_template(self, dhcp_server_ipv4, port_ips, hostname, cust
                                 "condition": "substring(relay4[1].hex, -{}, {}) == '{}'".format(class_len, class_len,
                                                                                                 client_class)
                             })
+
                     subnet_obj = {
-                        "id": dhcp_interface_name.replace("Vlan", ""),
+                        "id": 0 if smart_switch else dhcp_interface_name.replace("Vlan", ""),
```

Comment:
Can you provide a description on why this change is needed?

Reply:
updated PR description